conans/client/build/cmake.py

```diff
@@ -118,15 +118,28 @@ def is_multi_configuration(self):
 
     @property
     def command_line(self):
-        args = ['-G "%s"' % self.generator] if self.generator else []
-        if self.generator_platform:
-            if is_generator_platform_supported(self.generator):
-                args.append('-A "%s"' % self.generator_platform)
-            else:
-                raise ConanException('CMake does not support generator platform with generator '
-                                     '"%s:. Please check your conan profile to either remove the '
-                                     'generator platform, or change the CMake generator.'
-                                     % self.generator)
+        if self.generator_platform and not is_generator_platform_supported(self.generator):
```

Comment:
I think this went a little too far. I would change the ``cmake_flags.py`` helpers, but here in ``cmake.py`` I would try to keep with the minimal changes, in this case, ``self.generator`` can still be the composition of both, and use ``-G`` only whenever possible. 

Take into account that users might be taking the output of ``cmake.command_line`` and doing replacement themselves, and that kind of things. The less we change on the surface, the better.

Reply:
I think this is the simplest way of implementing this, if we want to keep using `-G` with the platform (only supported by CMake to keep compatibility with versions prior to 3.1), then:

 * if `generator_platform` is `ARM64`, then keep using `-A` (it is not supported using `-G`)
 * if `generator_platform` is `Win32`, do not append it to any generator in `-G` (not sure if supported), but use `-A` for VS2019
 * if `generator_platform` is `x64`, use `-A x64` for VS2019, but append `Win64` to `-G` for any other version.
 * if `generator` is VS2019, then always use `-A`

It could be simplified a little bit (or make it a little bit more complex if we want to consider the CMake version), but IMO it is going to be much more difficult to understand

Reply:
Basically to maintain the exact current behavior and ``cmake.command_line`` output, it is just moving the complexity from ``cmake_flags`` to the ``cmake`` helper. The last 3 points above are basically use ``-A`` always for VS2019. 

I would also prefer to clean this, but I think it will be just another source of breaking someone, and we can live without doing it. The main reason was simplifying a bit the internals of the "toolchain" pull request, but changing the command line of the main ``CMake`` build helper for that seems a risk that we can avoid.

Reply:
Ok